test/libsolidity/semanticTests/literalSuffixes/suffix_in_new.sol

```diff
@@ -0,0 +1,9 @@
+function twice(uint x) pure suffix returns (uint) { return x * 2; }
+
+contract C {
+    function run() public pure returns (uint[] memory) {
+        return new uint[](5 twice);
```

Comment:
Why are you testing this separately :-)? Do you want to write every possible program that you can write with any kind of expression and replace that expression with a suffix call for all of those cases :-)?

Reply:
Well, good point. I do have it covered by the bigger test case already so I'll remove it.

But the reason why it's here in the first place is that when I took over the PR it had numerous bugs. One of them was that suffixed literals were treated as constants and allowed e.g. in array sizes. When I plugged that hole I also added comprehensive syntax tests, trying to stick suffixes in various contexts and checking what will happen. This is one of them. This is not a function call and not a built-in. It's a construct that's probably processed specially at some point, and I don't know all the details of that so IMO it was prudent to cover it too instead of just assuming that there can be absolutely nothing wrong here.